refnx/reduce/reduce.py

```diff
@@ -1098,6 +1131,363 @@ def splice_datasets(self, ds):
+            I01=raw01.processed_spectrum["m_spec"][0],
+            I10=raw10.processed_spectrum["m_spec"][0],
+            I11=raw11.processed_spectrum["m_spec"][0],
+        )
+        raw00.processed_spectrum["m_spec_polcorr"] = np.asarray(
+            [[val[0] for val in I00]]
+        )
+        raw01.processed_spectrum["m_spec_polcorr"] = np.asarray(
+            [[val[0] for val in I01]]
+        )
+        raw10.processed_spectrum["m_spec_polcorr"] = np.asarray(
+            [[val[0] for val in I10]]
+        )
+        raw11.processed_spectrum["m_spec_polcorr"] = np.asarray(
+            [[val[0] for val in I11]]
+        )
+    elif pm is None and mp is None:
+        print("No spin-flip channel detected. Assuming R+- = R-+ = 0")
+        mp = np.zeros_like(pp.processed_spectrum["m_spec"][0])
+        pm = np.zeros_like(pp.processed_spectrum["m_spec"][0])
+        raw00 = mm
+        raw01 = mp
+        raw10 = pm
+        raw11 = pp
+        I00, I01, I10, I11 = correct_POL_efficiencies(
+            pp.processed_spectrum["m_lambda"][0],
+            I00=raw00.processed_spectrum["m_spec"][0],
+            I01=raw01,
+            I10=raw10,
+            I11=raw11.processed_spectrum["m_spec"][0],
+        )
+        raw00.processed_spectrum["m_spec_polcorr"] = np.asarray(
+            [[val[0] for val in I00]]
+        )
+        raw01 = None
+        raw10 = None
+        raw11.processed_spectrum["m_spec_polcorr"] = np.asarray(
+            [[val[0] for val in I11]]
+        )
+    return raw00, raw01, raw10, raw11
+
+
+def _not_none(*arrays):
+    """
+    Returns input if input is not not None
+    """
+    return [array for array in arrays if array is not None]
+
+
+def correct_POL_efficiencies(
+    wavelength, I00=None, I01=None, I10=None, I11=None
+):
+    """
+    Applies matrix correction of polarisation
+    inefficiencies to the measured spectra.
+
+    Parameters
+    ----------
+    pp : PlatypusNexus object
+        Polarised spectra for ++ spin channel
+    pm : PlatypusNexus object
+        Polarised spectra for +- spin channel
+    mp : PlatypusNexus object
+        Polarised spectra for -+ spin channel
+    mm : PlatypusNexus object
+        Polarised spectra for -- spin channel
+
+    Returns
+    -------
+    I00, I01, I10, I11 : PlatypusNexus object
+    with the efficiency-corrected spectrum
+    """
+    # The values used below are from Thomas Saerbeck's initial
+    # commissioning of the PLATYPUS polarisation system.
+    # TODO: Create a workflow that automatically fits
+    # commissioning data with an efficiency function to be used
+    # in the polarised_correction routine.
+
+    # Define polariser efficiency as function of wavelength
+    p1a = 0.993
+    p1b = 0.57
+    p1c = 0.47
+    P1 = p1a - p1b * p1c ** (wavelength).astype("float")
+
+    # Define analyser efficiency as function of wavelength
+    p2a = 0.993
+    p2b = 0.57
+    p2c = 0.51
+    P2 = p2a - p2b * p2c ** (wavelength).astype("float")
```

Comment:
wavelength is unlikely not to be a float array. However the best way to do this is:

```
wavelength = np.asfarray(wavelength)
P2 = p2a - p2b * p2c**wavelength
```